### ISSUE TITLE
lower detect performance

### ISSUE THREAD

**Issue by ysinwell** [ybdai7/Backdoor-indicator-defense#7]
Hello, I made modifications to the file utils/yamls/indicator/params_vanilla_Indicator.yaml in the original project to make it conform to the vanilla backdoor attack with \alpha = 0.2 benign_lr = 0.1. The specific modifications are as follows:
```
< resumed_model: False
< save_on_round: [400, 800, 1200]
---
> ### General Setting
> resumed_model: Nov.15_23.44.44/saved_model_global_model_1200.pt.tar
> save_on_round: [400, 500, 600, 800, 1000, 1200, 1300, 1450, 1650]
31c32
< poisoned_end_round: 13000
---
> poisoned_end_round: 1460
42c43
< semantic: True
---
> semantic: False
44c45
< pixel_pattern: False
---
> pixel_pattern: True
84c85
< dirichlet_alpha: 0.9
---
> dirichlet_alpha: 0.2
88c89
< end_round: 1900
---
> end_round: 1660
```
After conducting multiple experiments, I found that the best result I could obtain was a True Positive Rate (TPR) of 400/450 = 88.9% (lower than 99.2% in paper). 
Could you tell me how to improve the result?
Moreover, it's worth mentioning that I found when \alpha = 0.2 benign_lr = 0.1, the accuracy of quite a number of benign clients was only 10% - 20%. In the case of \alpha = 0.2, should the learning rate be lowered?
Thanks!

**Comment by ysinwell** [comment]
here is my result :
```
aggregated_model:[0, 1, 1, 1, 1, 1, 1, 1, 1, 1]
correctly detected malicious clients:400/450,                     undetected malicious clients:50/450
correctly detected benign clients:3811/4050,                     misclassified benign clients:239/4050
current VWM detection threshold: 95
 
global model on round:1649 | benign acc:92.41, benign loss:0.25258327178955076
global model on round:1649 | poisoned acc:41.949999999999996, poisoned loss:3.5298123779296877
saving model: global_model
watermarking acc:54.6875, watermarking loss:1.3659525762001674, target label (3) wm acc:83.12
global model on round:1650 | benign acc:92.41, benign loss:0.2525832702636719
global model on round:1650 | poisoned acc:41.949999999999996, poisoned loss:3.529812548828125
```

**Comment by ybdai7** [comment]
Hi,

As you may notice in the paper, the evaluated poisoning length is 250 global rounds. So, for an evaluation of 450 global rounds, I think it is reasonable to have an 88\% TPR (also may have some initialization difference due to random seeds). But if you still want to have the TPR increased, you may want to decrease the detection threshold from 95 to maybe 90. As you may find the FPR in your evaluation is less than 10%, which is also lower than the number in the paper (15%). Thus, it is reasonable for you to decrease the detection threshold to include more backdoor updates.

For the lr question, I think it is reasonable for clients to only have 10-20% accuracy under alpha=0.2. This is because, under such a highly non-iid data-partition method, some clients may only have a limited number of training data. And their training datasets are highly biased, and may only contain samples from a few classes. Locally training with such a dataset could result in only 10%-20% ACC, and you can see that the final ACC for the global model is good.  

I think it is also reasonable for you to decrease the lr. This is not a big issue as long as the final global model performs well in the main task.
